pkg/vault/test/foundry/fuzz/Swap.medusa.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { Rounding } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+
+import { BasePoolMath } from "../../../contracts/BasePoolMath.sol";
+import { BalancerPoolToken } from "../../../contracts/BalancerPoolToken.sol";
+
+import "../utils/BaseMedusaTest.sol";
+
+contract SwapMedusaTest is BaseMedusaTest {
+    uint256 internal constant MIN_SWAP_AMOUNT = 1e6;
+
+    int256 internal invariantDiff = 0;
+
+    constructor() BaseMedusaTest() {
+        vault.manuallySetSwapFee(address(pool), 0);
+    }
+
+    function optimize_currentInvariant() public view returns (int256) {
+        return -int256(invariantDiff);
+    }
+
+    function property_currentInvariant() public returns (bool) {
+        return invariantDiff >= 0;
```

Comment:
This will only measure the invariantDiff in the last swap, so it won't direct the sequence of calls to a sequence that decreases the invariant of the pool. I'd recommend measuring the invariant of the pool in the constructor, and then measuring the last invariant of the pool here, checking if the final invariant is not smaller than the first one. Something like:

```
    uint256 currentInvariant = computeInvariant();
    return currentInvariant >= firstInvariant;
```